src/realm/object-store/c_api/app.cpp

```diff
@@ -801,4 +801,18 @@ RLM_API char* realm_user_get_custom_data(const realm_user_t* user) noexcept
     return nullptr;
 }
 
+RLM_API char* realm_user_get_access_token(const realm_user_t* user)
+{
+    return wrap_err([&] {
+        return duplicate_string((*user)->access_token());
```

Comment:
Should the strings be duplicated here? I am not sure what the rule is on when we return a string that should be freed by the caller and when not. `realm_user_get_identity` just returns a pointer to the data held by the user object.

Reply:
If we can return a pointer to a data member we should that. But here these are computed strings, not members, so we need to copy them on the heap for the caller to release anyway.

Reply:
I think we need to return a dup here, since` access_token ` returns a new string (computed when it is called), thus returning a `const char*` could result in spurious crashes since we would be returning a reference to a local object (supposedly going to be destroyed as soon as the function ends)... sometimes the memory is still there, but eventually the string will have to be copied. 
So in this case it is better if we dup the string in the C API.

If the code makes sense, can we please approve and PR?

Reply:
@jedelbo  @fealebenpae

Reply:
The implementation makes sense to me, and I think we should merge.